Src/Base/AMReX_GpuControl.H

```diff
@@ -7,8 +7,8 @@
 
 #include <utility>
 
-#ifndef AMREX_GPU_MAX_THREADS
-#define AMREX_GPU_MAX_THREADS 256
+#ifndef AMREX_USE_GPU
+#define AMREX_GPU_MAX_THREADS 0
 #endif
```

Comment:
This is a bit unusual, since `AMREX_GPU_MAX_THREADS` should be defined by `#include <AMReX_Config.H>` at this point, no?

Reply:
`AMREX_GPU_MAX_THREADS` is only defined in `AMReX_Config.H` for GPU builds.  The problem is in a few places we have things like this in the code for both gpu and cpu. https://github.com/AMReX-Codes/amrex/blob/2ad56772eac9095b9678de5ced1e5440c4e579c9/Src/Base/AMReX_CTOParallelForImpl.H#L192
For the CPU version, the template parameter is simply ignored.

Reply:
Anyway, there are multiple ways to handle this.
1. Commit your suggestion.  But it will break all CPU builds.  So we will need to fix that.
2. Revert the change to the original code,
```
#ifndef AMREX_GPU_MAX_THREADS
#define AMREX_GPU_MAX_THREADS 256
#endif
```
3. The current way in the PR.
4. Always define AMREX_GPU_MAX_THREADS in the build systems.

Reply:
I think as a quick fix, we should go with:
> Always define AMREX_GPU_MAX_THREADS in the build systems.

in `AMReX_Config.H` and then do a fix later.

Below this line,
https://github.com/AMReX-Codes/amrex/blob/48f5ffc629cfcded97f9214dabc551248ab11d26/Tools/CMake/AMReX_Config.H.in#L43
I would write for now:
```C++
#ifndef AMREX_GPU_MAX_THREADS
#   define AMREX_GPU_MAX_THREADS 0
#endif
```

Reply:
Sure, we can always define AMREX_GPU_MAX_THREADS.  However, I won't insert spaces between `#` and `define`.  @zingale won't approve it.  But seriously, in this case it's better not to use any spaces, because then the words in the two lines will align perfectly.

Reply:
Haha, sure.